### DISCUSSION TITLE
Rendering code blocks in pages with configured syntax highlighting

### DISCUSSION THREAD

**Discussion by janko** [facebook/docusaurus#9031] [Q&A] [answered]
I'm using the [rehype-pretty-code](https://github.com/atomiks/rehype-pretty-code) plugin for syntax highlighting, as Prism doesn't highlight Ruby code well.

```js
// docusaurus.config.js
module.exports = {
  // ...  
  presets: [
    [
      "@docusaurus/preset-classic",
      {
        docs: {
          // ...
          rehypePlugins: [
            [require("rehype-pretty-code"), { theme: "dracula-soft" }]
          ]
        },
        // ...
      }
    ]
  ]
  // ...
}
```

I would now like to render code snippets on my home page, but using the same syntax highlighting that's configured above for docs pages. I tried importing the markdown content as a MDX component and wrapping it in a `<MDXContent>` component:

```jsx
// src/pages/index.js

import React from 'react';
import Layout from '@theme/Layout';
import Demo from './_demo.mdx';
import MDXContent from '@theme/MDXContent';

export default () => {
  return (
    <Layout title="Home">
      // ...
      <MDXContent>
        <Demo />
      </MDXContent>
    </Layout>
  );
};
```

However, that seems to use Prism and not my configured syntax highlighter.

**Comment by Josh-Cena** [comment]
`rehype-pretty-code` is, as other name implies, a Rehype plugin. As such it's only used in the pipeline of Markdown->JSX. However, the Prism approach is able to render anything that's already in JSX. A way to make it work is by putting the code in a Markdown file and importing it:

````md
<!-- front-page-code.md -->
```ruby
puts "Hello world"
```
````

```jsx
// front-page.js
import Code from "./front-page-code.md";

export default function Page() {
  return (
    // ...
    <Code />
  );
}
```

**Reply by janko** [reply]
> As such it's only used in the pipeline of Markdown->JSX.

What does this mean exactly? Why isn't Rehype used when importing MDX in custom pages? Is there a way to reuse the same pipeline that markdown docs use to get rendered?

> However, the Prism approach is able to render anything that's already in JSX.

That gives me the same result as importing a `.mdx` file. This doesn't even seem to use Prism, I needed to use the `<MDXContent>` component for that. However, as I said, I would like to avoid using Prism.

**Reply by slorber** [reply, marked as answer]
> What does this mean exactly? Why isn't Rehype used when importing MDX in custom pages? Is there a way to reuse the same pipeline that markdown docs use to get rendered?

It should, but have you declared your rehype plugin to pages as well? It's not in your config snippet.

```tsx
docs: {
          rehypePlugins: [
            [require("rehype-pretty-code"), { theme: "dracula-soft" }]
          ]
},
blog: {
          rehypePlugins: [
            [require("rehype-pretty-code"), { theme: "dracula-soft" }]
          ]
},
pages: {
          rehypePlugins: [
            [require("rehype-pretty-code"), { theme: "dracula-soft" }]
          ]
},
```

**Reply by janko** [reply]
I completely missed pages configuration, that worked perfectly, thank you! 🙌🏻
